anaSbitReadout.py

```diff
@@ -0,0 +1,376 @@
+
+    # loop over all branch names but the first (evnt num)
+    for branch in brName[1:]:
+        print "Unpacking now", branch + \
+            ",", str(rawData[branch].shape).translate(None, "(),"), "entries"
+        # h is a dummy variable to be printed in case of error
+        for h, word in enumerate(rawData[branch]):
+
+            sbitAddr = ((word) & 0x7FF)
+            # INVALID ADDRESS CHECK
+            if sbitAddr >= 1536:
+                continue
+
+            vfatN[0] = (7 - sbitAddr/192 + ((sbitAddr % 192)/64)*8)
+            sbitSize[0] = ((word >> 11) & 0x7)
+            chHitPerCluster[0] = 2*(sbitSize[0]+1)
+            L1Delay[0] = ((word >> 14) & 0xFFF)
+            eta = vfat_to_etaphi[vfatN[0]][0]
+            phi = vfat_to_etaphi[vfatN[0]][1]
+            # SBIT always includes doublet of adjacent channels
+            vfatCH[0] = 2*(sbitAddr % 64)
+            vfatCH[1] = vfatCH[0] + 1
+            strip[0] = vfat_ch_strips[vfatN[0]]['Strip'][vfatCH[0]]
+            strip[1] = vfat_ch_strips[vfatN[0]]['Strip'][vfatCH[1]]
+            # In case of wrong mapping adjacent channels may not be adjacent strips, which is physically inconsistent
+            if(np.abs(strip[0] - strip[1]) > 1):
+                print 'WARNING: not adjacent strips'
+                time.sleep(3)
+
+            # filling vfat 1Dhistos
+            vfat_h_strip[vfatN[0]].Fill(strip[0])
+            vfat_h_strip[vfatN[0]].Fill(strip[1])
+            vfat_h_ch[vfatN[0]].Fill(vfatCH[0])
+            vfat_h_ch[vfatN[0]].Fill(vfatCH[1])
+            vfat_h_delay[vfatN[0]].Fill(L1Delay[0])
+            vfat_h_sbitSize[vfatN[0]].Fill(sbitSize[0])
+
+            # filling ieta 1Dhistos
+            ieta_h_strip[eta].Fill((phi-1)*128+strip[0])
+            ieta_h_strip[eta].Fill((phi-1)*128+strip[1])
+            ieta_h_ch[eta].Fill((phi-1)*128+vfatCH[0])
+            ieta_h_ch[eta].Fill((phi-1)*128+vfatCH[1])
+            ieta_h_delay[eta].Fill(L1Delay[0])
+            ieta_h_sbitSize[eta].Fill(sbitSize[0])
+
+            # filling 2Dhisto
+            hh_ieta_strip[0].Fill(strip[0]+128*(phi-1), eta)
+            hh_ieta_strip[0].Fill(strip[1]+128*(phi-1), eta)
+            hh_ieta_ch[0].Fill(vfatCH[0]+128*(phi-1), eta)
+            hh_ieta_ch[0].Fill(vfatCH[1]+128*(phi-1), eta)
+
+            """
+            A single sbit word can specify up to 16 adjacent channel hits
+            The following loop takes care of this possibility
+            """
+            for i in range(2, chHitPerCluster[0]):
+                # filling the adjacent channel
+                vfatCH[i] = vfatCH[i-1] + 1
+                # if the new channel exceeds the total VFAT channels, increase phi and move to the first cannel of the next VFAT
+                if vfatCH[i] >= 128 and phi < 3:
+                    phi = phi + 1
+                    vfatCH[i] = 0
+                    vfatN[0] = etaphi_to_vfat[eta][phi]
```

Comment:
this is redundant; you already know `vfatN[0]` from line 194 (i.e. the for loop at line 236 is inside the loop at 187 so `vfatN[0]` is still the same value.  This also means the dictionary `etaphi_to_vfat` is not needed (since this is the only place it is used).